clang/tools/clang-scan-deps/ClangScanDeps.cpp

```diff
@@ -472,7 +465,8 @@ class FullDeps {
             JOS.attributeArray("command-line",
                                toJSONStrings(JOS, MD.getBuildArguments()));
             JOS.attribute("context-hash", StringRef(MD.ID.ContextHash));
-            JOS.attributeArray("file-deps", toJSONSorted(JOS, MD.FileDeps));
+            // TODO: Shuffle these to prevent tests from depending on the order.
```

Comment:
Because we might want to change one arbitrary deterministic ordering to another arbitrary deterministic ordering and I'd like for the tests to be resilient against that so that they don't need to change every time we do that.

Reply:
It's not clear to me the value of being able to change the order (I don't think this would happen often) outweighs the benefits of better test coverage for deterministic output.

Regardless, I don't want to random shuffling the output of `clang-scan-deps`, because it breaks diffing the output.  To me this TODO just seems like a bad idea.

Reply:
Hmm, I wanted to seed the random shuffle with something "stable" (like the repo revision number) to be able to diff in tests. I also wanted to add a `clang-scan-deps` flag to disable it entirely for development purposes. Is that something that you still consider a bad idea?